src/main/java/com/jacob/duke/Storage.java

```diff
@@ -0,0 +1,228 @@
+package main.java.com.jacob.duke;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.DataInputStream;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import main.java.com.jacob.duke.task.Deadline;
+import main.java.com.jacob.duke.task.Event;
+import main.java.com.jacob.duke.task.Task;
+import main.java.com.jacob.duke.task.Todo;
+
+
+
+
+
+public class Storage {
+    private StringBuffer stringBufferOfData = new StringBuffer();
+    private String filename;
+
+    /**
+     * Constructor for the Storage object
+     * @param filename path of the file to be accessed
+     */
+    public Storage(String filename) {
+        this.filename = filename;
+        accessTaskListInFileSystem(getCurrentDirectory());
+    }
+
+    //get current directory
+    private String getCurrentDirectory() {
+        return System.getProperty("user.dir");
+    }
+
+    //Access the current list, creates the folder and files if they do not exist
+    private void accessTaskListInFileSystem(String current) {
+        String[] fileParents = filename.split("/");
+        String parent = fileParents[0];
+        java.nio.file.Path directoryPath = java.nio.file.Paths.get(current, parent);
+        boolean directoryExists = java.nio.file.Files.exists(directoryPath);
+
+
```

Comment:
There's one extra space here, not sure if it is a violation but you might want to delete it so that your code will look more consistent throughout? There is one extra space at line 65 too